spdx/file.py

```diff
@@ -202,29 +212,40 @@ def validate_file_types(self, messages):
                 messages.append(f"{file_type} is not of type FileType.")
         return messages
 
-    def validate_checksum(self, messages):
-        if not isinstance(self.chksum, checksum.Algorithm):
-            messages.append(
-                "File checksum must be instance of spdx.checksum.Algorithm"
-            )
-        else:
-            if not self.chksum.identifier == "SHA1":
-                messages.append("File checksum algorithm must be SHA1")
+    def validate_checksums(self, messages: ErrorMessages):
+        for checksum in self.checksums.values():
+            if not isinstance(checksum, Checksum):
+                messages.append("File checksum must be instance of spdx.checksum.Checksum.")
+
+        if self.get_checksum(ChecksumAlgorithm.SHA1) is None:
+            messages.append("At least one file checksum algorithm must be SHA1")
 
-        return messages
 
-    def calc_chksum(self):
+    def calculate_checksum(self, hash_algorithm='SHA1'):
+        if hash_algorithm not in ChecksumAlgorithm.__members__:
+            raise ValueError
         BUFFER_SIZE = 65536
 
-        file_sha1 = hashlib.sha1()
+        file_hash = hashlib.new(hash_algorithm.lower())
         with open(self.name, "rb") as file_handle:
             while True:
                 data = file_handle.read(BUFFER_SIZE)
                 if not data:
                     break
-                file_sha1.update(data)
+                file_hash.update(data)
+
+        return file_hash.hexdigest()
+
+    def get_checksum(self, hash_algorithm: ChecksumAlgorithm = ChecksumAlgorithm.SHA1) -> Checksum:
+        return self.checksums[hash_algorithm]
 
-        return file_sha1.hexdigest()
+    def set_checksum(self, new_checksum: Checksum):
+        if not isinstance(new_checksum, Checksum):
+            raise SPDXValueError
+        if new_checksum.identifier in self.checksums:
+            self.checksums[new_checksum.identifier].value = new_checksum.value
```

Comment:
nit: I would prefer replacing the entry instead of mutating the current one

Reply:
fixed